src/client/datascience/jupyter/jupyterNotebookProvider.ts

```diff
@@ -20,21 +21,40 @@ export class JupyterNotebookProvider implements IJupyterNotebookProvider {
     constructor(@inject(IJupyterServerProvider) private readonly serverProvider: IJupyterServerProvider) {}
 
     public async disconnect(options: ConnectNotebookProviderOptions): Promise<void> {
-        const server = await this.serverProvider.getOrCreateServer(options);
-
-        return server?.dispose();
+        const ui = new DisplayOptions(options.disableUI === true);
+        try {
+            const server = await this.serverProvider.getOrCreateServer({
+                getOnly: false,
+                ui,
+                resource: options.resource,
+                token: options.token
+            });
+            return server?.dispose();
+        } finally {
+            ui.dispose();
+        }
     }
 
     public async connect(options: ConnectNotebookProviderOptions): Promise<IJupyterConnection | undefined> {
-        const server = await this.serverProvider.getOrCreateServer(options);
-        return server?.getConnectionInfo();
+        const ui = new DisplayOptions(options.disableUI === true);
+        try {
+            const server = await this.serverProvider.getOrCreateServer({
+                getOnly: false,
+                ui: new DisplayOptions(options.disableUI === true),
```

Comment:
This seems like a typo. Shouldn't it use the variable?

Reply:
No, connect is one of those weird methods that can be called to pre-warm. 
There are juts two public methods that can be called `connect` & `kernel.start`.

I'm still not sure what should be done here. When I work on displaying the UI, i'll ensure to test this & make the changes accordingly.